sdk/textanalytics/test-resources.json

```diff
@@ -0,0 +1,89 @@
+            "type": "string",
+            "defaultValue": "[resourceGroup().name]",
+            "metadata": {
+                "description": "THe base resource name."
+            }
+        },
+        "testApplicationOid": {
+            "type": "String",
+            "metadata": {
+                "description": "The principal to assign the role to. This is application object id."
+            }
+        },
+        "tenantId": {
+            "type": "String",
+            "metadata": {
+                "description": "The tenant ID to which the application and resources belong."
+            }
+        },
+        "testApplicationId": {
+            "type": "String",
+            "metadata": {
+                "description": "The application client ID used to run tests."
+            }
+        },
+        "testApplicationSecret": {
+            "type": "String",
+            "metadata": {
+                "description": "The application client secret used to run tests."
+            }
+        }
+    },
+    "variables": {
+        "taRoleId": "[concat('/subscriptions/', subscription().subscriptionId, '/providers/Microsoft.Authorization/roleDefinitions/a97b65f3-24c7-4388-baec-2e87135dc908')]"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Authorization/roleAssignments",
+            "apiVersion": "2018-09-01-preview",
+            "name": "[guid(concat(variables('taRoleId'), parameters('baseName')))]",
+            "dependsOn": [
+                "[parameters('baseName')]"
+            ],
+            "properties": {
+                "principalId": "[parameters('testApplicationOid')]",
+                "roleDefinitionId": "[variables('taRoleId')]"
+            }
+        },
+        {
+            "type": "Microsoft.CognitiveServices/accounts",
+            "name": "[parameters('baseName')]",
+            "apiVersion": "2017-04-18",
+            "sku": {
+                "name": "S"
+            },
+            "kind": "TextAnalytics",
+            "location": "[resourceGroup().location]",
+            "properties": {
+                "customSubDomainName": "[parameters('baseName')]"
+            }
+        }
+    ],
+    "outputs": {
+        "AZURE_TENANT_ID": {
+            "type": "String",
+            "value": "[parameters('tenantId')]"
+        },
+        "AZURE_CLIENT_ID": {
+            "type": "String",
+            "value": "[parameters('testApplicationId')]"
+        },
+        "AZURE_CLIENT_SECRET": {
+            "type": "String",
+            "value": "[parameters('testApplicationSecret')]"
+        },
+        "AZURE_TEXT_ANALYTICS_SUBSCRIPTION_KEY": {
+            "type": "string",
+            "value": "[listKeys(resourceId('Microsoft.CognitiveServices/accounts', parameters('baseName')), '2017-04-18').key1]"
+        },
+        "AZURE_TEXT_ANALYTICS_ENDPOINT": {
+            "type": "string",
+            "value": "[concat('https://', parameters('baseName'), '.cognitiveservices.azure.com/')]"
```

Comment:
I would make this a variable in the template rather than creating it here. It makes it easier to follow because we just have to jump to the top of this script rather than ctrl+f through all our concats.